_examples/producer/producer.go

```diff
@@ -58,7 +59,6 @@ func setupCloseHandler(exitCh chan struct{}) {
 
 func publish(ctx context.Context, publishOkCh <-chan struct{}, confirmsCh chan<- *amqp.DeferredConfirmation, confirmsDoneCh <-chan struct{}, exitCh chan struct{}) {
 	config := amqp.Config{
-		Vhost:      "/",
```

Comment:
Does other example code use `Vhost:` when it's not necessary?

Reply:
My suggestion is that the configured uri vhost should be used first. In other cases, you can configure it yourself.
Of course it's just my suggestion.

Reply:
https://github.com/rabbitmq/amqp091-go/pull/238#issuecomment-1881297418

I'm not sure I understand the benefit of this. IMO, having `Vhost` field set in the config is very clear, and it's better than relying in the implicit behaviour of URI parsing.

Reply:
> [#238 (comment)](https://github.com/rabbitmq/amqp091-go/pull/238#issuecomment-1881297418)
> 
> I'm not sure I understand the benefit of this. IMO, having `Vhost` field set in the config is very clear, and it's better than relying in the implicit behaviour of URI parsing.

The reason why I modified it like this is because after I set the Vhost in the uri, I ran the code and reported an error. After comparison, I found that this was not the vhost I set.

Reply:
Maybe this is just my personal understanding. If you think the original way is better, I think my submission can be canceled. At the same time, I am also happy to communicate with you, thank you.